accepted/2020/workloads/workload-resolvers.md

```diff
@@ -0,0 +1,237 @@
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net5.0-android42</TargetFramework>
+  </PropertyGroup>
+
+</Project>
+```
+
+## Workload guidance
+
+The bulk of functionality from workloads will be contained in workload "packs", which will be the unit of delivery for workloads.  This will include MSBuild targets and tasks.  The assets from the packs will be laid out under the dotnet root folder, probably in `packs/<Pack ID>/<Pack Version>` or a similar folder.
+
+Workloads will also provide a manifest package which describes which packs are necessary for which workloads.  The manifest will specify the name and version for each pack that is available for a given .NET SDK release band, as well as which packs are required for each logical workload.  The manifest for workloads will be present even when the workloads it describes are not installed.
+
+### Workload Manifest Targets Files
+
+In addition to the manifest file itself (likely in json format), the workload manifest package will include a `WorkloadManifest.targets` file which should import the necessary .targets files from the workload pack if a project depends on the workload.  The `WorkloadManifest.targets` files from all workload manifests will be imported into all projects that use Microsoft.NET.Sdk.
+
+The `WorkloadManifest.targets` should use MSBuild Sdk imports to import .targets files from an workload pack.  This should be done conditionally based on properties that determine whether the workload should be used.  For example, the Android workload manifest targets file might look like this:
+
+```xml
+<Import Project="Sdk.targets" Sdk="Xamarin.Android.Sdk"
+        Condition="'$(TargetPlatformIdentifier)' == 'android'" />
+```
+
+This indicates that the `Sdk.targets` file should be imported from the `Xamarin.Android.Sdk` workload pack if the `TargetPlatformIdentifier` is `android`.  The version of the `Xamarin.Android.Sdk` workload pack to use will be the one defined in the workload manifest json file.
+
+Because the workload manifest targets files will be imported by all projects, it is important to try to guard against bugs in these targets from causing issues with projects that don't intend to use the workload at all.  One way of reducing risk can be to split the logic into multiple targets files, where the targets files with more complex logic are only imported from the entry point workload manifest targets file based on a simple condition.  For example:
+
+```xml
+<!-- WorkloadManifest.targets (entry point): -->
+
+<Project>
+    <Import Project="Xamarin.Manifest.Android.Framework.targets" Condition="'$(TargetPlatformIdentifier)'=='android'" />
```

Comment:
The condition in uniform in the example here and the one above. I'm not an MSBuild person, but it is hard for me to understand the value of this second option if the condition is constant.

Reply:
If you start to have complex logic to determine which workload packs are imported, then this pattern helps reduce risk that bugs in that logic will impact projects which don't use your workload at all.

Reply:
It sounds like the conditions might become more complex over time (even within this release).

I don't have anything to offer here. Just a question.